pkg/cluster/private_registry.go

```diff
@@ -34,14 +34,32 @@ func GetPrivateRepo(cluster *v3.Cluster) *rketypes.PrivateRegistry {
 	return nil
 }
 
+// TransformProvV2RegistryCredentialsToDockerConfigJSON transforms a ProvV2 registry secret into a credentialprovider.DockerConfigJSON secret
+func TransformProvV2RegistryCredentialsToDockerConfigJSON(url string, registryCredentials *corev1.Secret) (*corev1.Secret, error) {
+	username := string(registryCredentials.Data["username"])
+	password := string(registryCredentials.Data["password"])
+	authConfig := credentialprovider.DockerConfigJSON{
+		Auths: credentialprovider.DockerConfig{
+			url: credentialprovider.DockerConfigEntry{
+				Username: username,
+				Password: password,
+			},
+		},
+	}
+	j, err := json.Marshal(authConfig)
+	return &corev1.Secret{
```

Comment:
Either that or actually make the secret a secret.

Reply:
I had thought about creating a new secret, but felt that it would be duplicating information that we already have in other secrets which is why I opted not to, and instead created one in memory. I am also worried about potential regressions within RKE1, since the two share similar code paths when deploying the agent, and so I tried to keep refactoring of existing behavior to a minimum. I can make this change, but private registry behavior for RKE1 will have to be revalidated as well.

Reply:
Okay, I've changed things up to just use a `map[string][]byte` instead of the actual secret. I've done a quick smoke test and things seem to be working as expected, let me know if there are other changes related to the refactoring